Runtime/JSNativeApi.cs

```diff
@@ -502,17 +502,38 @@ public static unsafe JSValue Wrap(this JSValue wrapper, object value, out JSRefe
         return wrapper;
     }
 
+    public static bool TryUnwrap(this JSValue thisValue, out object? value)
+    {
+        napi_status status = napi_unwrap(Env, (napi_value)thisValue, out nint result);
+        if (status == napi_status.napi_invalid_arg)
+        {
+            value = null;
+            return false;
+        }
+
+        status.ThrowIfFailed();
+        value = GCHandle.FromIntPtr(result).Target;
+        return true;
+    }
 
-    public static object? Unwrap(this JSValue thisValue)
+    public static object Unwrap(this JSValue thisValue)
     {
         napi_unwrap(Env, (napi_value)thisValue, out nint result).ThrowIfFailed();
-        return GCHandle.FromIntPtr(result).Target;
+        return GCHandle.FromIntPtr(result).Target!;
     }
 
-    public static object? RemoveWrap(this JSValue thisValue)
+    public static bool RemoveWrap(this JSValue thisValue, out object? value)
```

Comment:
>RemoveWrap [](http://example.com/codeflow?start=23&length=10)

Should we name it TryRemoveWrap?

Reply:
Maybe, but I didn't because I expect most callers will ignore the return value because they don't care if the remove failed because it was never wrapped or already removed. It's similar to how `bool ICollection<T>.Remove(T item)` is not named `TryRemove`.